onnx/defs/tensor/defs.cc

```diff
@@ -257,23 +258,27 @@ ONNX_OPERATOR_SET_SCHEMA(
           // Type inference
           propagateElemTypeFromInputToOutput(ctx, 0, 0);
           // Shape Inference if 2nd input data (the target shape) is available
+          // or the target shape is generated via partial data propagation
           const TensorProto* targetShapeInitializer = ctx.getInputData(1);
-          if (!targetShapeInitializer) {
+          const auto* shapeInput = ctx.getShapeInput(1);
+
+          if (!targetShapeInitializer && !shapeInput) {
             return;
           }
+
+          // The targetShapeProto represents the specified shape for output.
+          TensorShapeProto targetShapeProto;
           int allowzero = static_cast<int>(getAttribute(ctx, "allowzero", 0));
-          // Make targetShape (0 -> same as originalShape, -1 -> inferred).
-          // The targetShape vector represents the specified shape for output.
-          std::vector<int64_t> targetShape;
-          if (targetShapeInitializer->has_raw_data()) {
-            const std::string& bytes = targetShapeInitializer->raw_data();
-            targetShape.insert(
-                targetShape.end(),
-                reinterpret_cast<const int64_t*>(bytes.c_str()),
-                reinterpret_cast<const int64_t*>(bytes.c_str() + bytes.size()));
-          } else {
-            const auto& data = targetShapeInitializer->int64_data();
-            targetShape.insert(targetShape.end(), data.begin(), data.end());
+
+          if (targetShapeInitializer) {
+            auto targetShape = ParseData<int64_t>(targetShapeInitializer);
+            for (auto val : targetShape) {
+              targetShapeProto.add_dim()->set_dim_value(val);
+            }
+          }
+
+          if (shapeInput) {
```

Comment:
Perhaps else if for (shapeInput) ?

I thought targetShapeInitializer and shapeInput cannot be valid at the same time.